### ISSUE TITLE
Live reload keeps adding in connections

### ISSUE THREAD

**Issue by unixcow** [infinitered/reactotron#1154]
Enabling live reload in dev menu for a project keeps adding a new connections to reactotron without even clearing the old one. This is so annoying 😅 and i don't know if it's supposed to be like that. It seems like a bug for me 

![image](https://user-images.githubusercontent.com/50556983/65544223-17e88d80-df13-11e9-9910-7ebe1255dbfa.png)

## Reproduction

Just connect your device or emulator (in my case an android device), enable live reload _(because hot reload sucks and doesn't work most of the time 😂)_. Now every time you save a change and it live reloads, reactotron will add a new connection.

**Comment by rmevans9** [comment]
Do you have async storage configured with reactotron?

**Comment by unixcow** [comment]
@rmevans9 yeah

**Comment by unixcow** [comment]
this is my config, i believe i don't have async storage enabled.

```js
reactron
    .configure({ host: '192.168.1.20', name: "7RU3L1F3" })
    .useReactNative({
			overlay: false,
		})
    .connect()
```

**Comment by unixcow** [comment]
@rmevans9 this is my config, i believe i don't have async storage enabled.

```js
reactron
    .configure({ host: '192.168.1.20' })
    .useReactNative({
			overlay: false,
		})
    .connect()
```

**Comment by PR1YANKPAT3L** [comment]
I'm also experiencing the same issue. Below is my configuration:

```
Reactotron
    .configure({
        host: "192.168.x.xx",
    })
    .useReactNative({
        asyncStorage: false,
        networking: {
            ignoreUrls: /symbolicate/,
        },
        editor: false,
        errors: { veto: (stackFrame) => false },
        overlay: false,
    })
    // .use(middleware)
    .connect();
```

**Comment by yuritoledo** [comment]
Any suggestion?

**Comment by yuritoledo** [comment]
@unixcow @PR1YANKPAT3L try something like this, worked for me:

```
import Reactotron from 'reactotron-react-native'
import AsyncStorage from '@react-native-community/async-storage'
Reactotron
  .useReactNative()
  .configure()
  .setAsyncStorageHandler(AsyncStorage) // <- here!
  .connect()

```

**Comment by unixcow** [comment]
well, my flooded reactotron thanks you sir. This should be either mentioned in the docs or marked as a bug though.

**Comment by yuritoledo** [comment]
I agree with you and as soon as possible flagged as a bug.

@rmevans9 what you can do about this?

**Comment by rmevans9** [comment]
:wave: Sorry guys. Yes, we need to get that added to the docs ASAP. It was mentioned in the breaking changes of reactotron-react-native but that obviously doesn't help new people coming to Reactotron. I have been a bit busy lately but I will try and get some time to get the docs updated.

**Comment by rmevans9** [comment]
Actually looking at our current docs:

https://github.com/infinitered/reactotron/blob/master/docs/quick-start-react-native.md#configure-reactotron-with-your-project

It seems like I did get this line added. I am unsure what you guys are hoping to have flagged as a bug since it is mentioned in the setting up docs?

**Comment by unixcow** [comment]
@rmevans9 maybe a hint that this line isn't optional and will result in this unexpected behavior if not added?

**Comment by yuritoledo** [comment]
Thank you sir, for your quick attention.

For me its ok now, with this commit: https://github.com/infinitered/reactotron/commit/f9e2c925685ebe8145d70e4448314c963d8a2798

**Comment by PR1YANKPAT3L** [comment]
@yuritoledo I will check it out later tonight, thanks for the reply with a fix.

**Comment by yuritoledo** [comment]
@PR1YANKPAT3L Cool!

**Comment by rmevans9** [comment]
@unixcow We can certainly do that. A warning might make a lot more sense then just trying to default it out. I am also trying to think of a way to not have to depend on AsyncStorage but due to an issue on Android I am unable to do that so far.

**Comment by unixcow** [comment]
@rmevans9 yeah because if it weren't for my reporting about it i would've just thought that it's broken. imho, Ideally AsyncStorage should be disabled unless explicitly enabled by the user at that time he/she will have to look into the docs and find out that they need to pass it to reactotron to avoid unexpected behavior. If disabling by default isn't an option though, at least make it error out pointing to the docs.

**Comment by frederikocmr** [comment]
> @unixcow @PR1YANKPAT3L try something like this, worked for me:
> 
> ```
> import Reactotron from 'reactotron-react-native'
> import AsyncStorage from '@react-native-community/async-storage'
> Reactotron
>   .useReactNative()
>   .configure()
>   .setAsyncStorageHandler(AsyncStorage) // <- here!
>   .connect()
> ```
> 
> [#770 (comment)](https://github.com/infinitered/reactotron/issues/770#issuecomment-522484091)

Worked for me just by adding `.setAsyncStorageHandler(AsyncStorage)`, as stated in the docs:
" AsyncStorage would either come from `react-native` or `@react-native-community/async-storage` depending on where you get it from"

**Comment by JB-CHAUVIN** [comment]
Same problem goes for me...

**Comment by PR1YANKPAT3L** [comment]
> @unixcow @PR1YANKPAT3L try something like this, worked for me:
> 
> ```
> import Reactotron from 'reactotron-react-native'
> import AsyncStorage from '@react-native-community/async-storage'
> Reactotron
>   .useReactNative()
>   .configure()
>   .setAsyncStorageHandler(AsyncStorage) // <- here!
>   .connect()
> ```
> 
> [#770 (comment)](https://github.com/infinitered/reactotron/issues/770#issuecomment-522484091)

I was away for a while, so couldn't test. Adding AsyncStorage resolved the issue. Thanks again!

**Comment by lucasmathioni** [comment]
You can define a unique ID for the device. This way you will establish a single connection per device.
```
Reactotron.configure({
  getClientId: () => UNIQUE_ID,
})```

**Comment by duspada** [comment]
> You can define a unique ID for the device. This way you will establish a single connection per device, without the need to add async storage.
> 
> ```
> Reactotron.configure({
>   getClientId: () => UNIQUE_ID,
> })
> ```

worked!!!
like this:


`import DeviceInfo from 'react-native-device-info';

  const tron = Reactotron.configure({
    host: scriptHostname,
    getClientId: () => DeviceInfo.getAndroidId(), // fix multiple connections
  })`

**Comment by joshuayoes** [comment]
@morganick 👋

**Comment by AldoMX** [comment]
> You can define a unique ID for the device. This way you will establish a single connection per device, without the need to add async storage.
> 
> ```
> Reactotron.configure({
>   getClientId: () => UNIQUE_ID,
> })
> ```

There is already a unique id in the `expo-constants` package, the `installationId`:

```typescript
import Constants from 'expo-constants';

Reactotron.configure({
  getClientId: async () => Constants.installationId
});
```

[Docs](https://docs.expo.dev/versions/latest/sdk/constants/#nativeconstants) say you should generate and store your own id instead, but I believe this is a legitimate use case for the installation id 😅.
